### PR TITLE
Add autofill test pages

### PR DESCRIPTION
**Pull request by GioSensation**
**Asana:** https://app.asana.com/0/72649045549333/1200085115126414/f
**Reviewer:** @jonathanKingston 
**CC:** @brindy 

## Description
Adds test pages for a few autofill cases to avoid regressions.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -55,6 +55,12 @@ <h2>Privacy Protections Tests</h2>
 	<li><a href='http://privacy-test-pages.glitch.me/privacy-protections/surrogates/'>Surrogates</a></li>
 </ul>
 
+<h2>Autofill</h2>
+<ul>
+	<li><a href="./autofill/modal.html">Autofill form within modal</a></li>
```
Comment (kdzwinel) [inline index.html:60]: Let's make it clear that it's an *email* autofill.
```suggestion
	<li><a href="./autofill/modal.html">Email autofill form within modal</a></li>
```

File: index.html [hunk 2]
```diff
@@ -55,6 +55,12 @@ <h2>Privacy Protections Tests</h2>
...
+	<li><a href="./autofill/form-submission.html">Form submission detection</a></li>
```
Comment (kdzwinel) [inline index.html:61]: As far as I understand this page will help to test both detection and autofill, so we can add that

```suggestion
	<li><a href="./autofill/form-submission.html">Form submission detection and autofill</a></li>
```

File: autofill/modal.html [hunk 3]
```diff
@@ -0,0 +1,51 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width">
+  <title>Self-closing modal test</title>
```
Comment (kdzwinel) [inline autofill/modal.html:7]: let's keep the names of pages from `index.html` in sync with titles

File: autofill/form-submission.html [hunk 4]
```diff
@@ -0,0 +1,88 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width">
+  <title>Login forms</title>
```
Comment (kdzwinel) [inline autofill/form-submission.html:7]: ditto, let's keep titles in sync with page names from `index.html`

**Review by kdzwinel — COMMENTED** [review]
Thanks @GioSensation. I found couple of nits re naming, but other than that it LGTM!

**Comment by GioSensation** [comment]
Thanks for the tips, @kdzwinel. Addressed and merged.

**Pull request merged**